Userland/Applications/Spreadsheet/CellSyntaxHighlighter.cpp

```diff
@@ -13,44 +14,43 @@ namespace Spreadsheet {
 
 void CellSyntaxHighlighter::rehighlight(Palette const& palette)
 {
-    auto text = m_client->get_text();
-    m_client->spans().clear();
-    if (!text.starts_with('=')) {
-        m_client->do_update();
-        return;
-    }
+    m_client->clear_spans();
+
+    if (m_client->get_text().starts_with('=')) {
 
-    JS::SyntaxHighlighter::rehighlight(palette);
+        JS::SyntaxHighlighter::rehighlight(palette);
 
-    // Highlight the '='
-    m_client->spans().empend(
-        GUI::TextRange { { 0, 0 }, { 0, 1 } },
-        Gfx::TextAttributes {
-            palette.syntax_keyword(),
-            Optional<Color> {},
+        // Highlight the '='
+        m_client->do_set_spans({ {
```

Comment:
This will reset the spans made by the JS highlighter and only highlight `=`.

Reply:
Yeah. My suggestion would be to copy the vector from `m_client->spans()` after running the JS highlighter, then append any extra spans to it, then set spans with that vector.